src/haddock/gear/config_reader.py

```diff
@@ -40,17 +40,34 @@
 # thanks https://stackoverflow.com/questions/39158902
 _sub_header_re = re.compile(r'^ *\[(\w+(?:\.\w+)+)\]')
 
-# https://regex101.com/r/q2fuFl/1
-_string_re = re.compile(r'''^ *(\w+) *= *("(.*?)"|'(.*?)')''')
+# string regular expression here matches the expectations of modules' defaults
+# there are no modules defaults that expect any other character. So, if we
+# block it's read from here, avoid also downstream errors.
+# https://regex101.com/r/TRMx0B/1
+_string_re = re.compile(
+    r'''^ *(\w+) *= *'''
+    r'''("([a-zA-Z0-9\.]*?)"|'([a-zA-Z0-9\.]*?)')'''
```

Comment:
Why not allowing for example `-` in a string?   Could be in a filename for example.

Reply:
So, that's the discussion I want to have. `-` is accepted on Paths but not strings because there are not the `-` character in string module's parameters. If we implement that only used chars can be input we correct for both rough bad input and malicious characters.